common/source/fs/BDVirtualDevice.cpp

```diff
@@ -73,6 +74,8 @@ int32 BDVirtualDevice::readData(uint32 offset, uint32 size, char *buffer)
       ArchInterrupts::disableInterrupts();
   }
 
+  assert((!currentThread || currentThread->isStackCanaryOK()) && "Kernel stack corruption detected.");
```

Comment:
Packaging this into a function probably would be a good idea now that it's used so often

Reply:
done